src/components/EditFreakForm/__test__/EditFreakForm.test.jsx

```diff
@@ -0,0 +1,204 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import EditFreakForm from '../EditFreakForm';
+import { FreakModelDefault, FreakModelKeys } from '../../../models/freaks';
+
+describe('EditFreakForm', () => {
+  const onChange = jest.fn();
+
+  beforeEach(() => {
+    onChange.mockReset();
+  });
+
+  it('should trigger onChange when the firstName is changed', () => {
+    render(
+      <EditFreakForm
+        onChange={ onChange }
+      />,
+    );
+
+    const firstNameInput = screen.getByTestId('first-name-input');
+    expect(firstNameInput).toBeInTheDocument();
+
+    fireEvent.change(firstNameInput, { target: { value: 'Marian' } });
+
+    expect(onChange).toBeCalledWith({
+      ...FreakModelDefault,
+      [FreakModelKeys.firstName]: 'Marian',
+    });
+  });
+
+  it('should trigger onChange when the lastName is changed', () => {
+    render(
+      <EditFreakForm
+        onChange={ onChange }
+      />,
+    );
+
+    const lastNameInput = screen.getByTestId('last-name-input');
+
+    fireEvent.change(lastNameInput, { target: { value: 'Badea' } });
+
+    expect(onChange).toBeCalledWith({
+      ...FreakModelDefault,
+      [FreakModelKeys.lastName]: 'Badea',
+    });
+  });
+
+  it('should trigger onChange when the email is changed', () => {
+    render(
+      <EditFreakForm
+        onChange={ onChange }
+      />,
+    );
+
+    const emailInput = screen.getByTestId('email-input');
+
+    fireEvent.change(emailInput, { target: { value: 'badea@email.com' } });
+
+    expect(onChange).toBeCalledWith({
+      ...FreakModelDefault,
+      [FreakModelKeys.email]: 'badea@email.com',
+    });
+  });
+
+  it('should trigger onChange when the description is changed', () => {
+    render(
+      <EditFreakForm
+        onChange={ onChange }
+      />,
+    );
+
+    const descriptionInput = screen.getByTestId('description-input');
+
+    fireEvent.change(descriptionInput, { target: { value: 'I am Marian' } });
+
+    expect(onChange).toBeCalledWith({
+      ...FreakModelDefault,
+      [FreakModelKeys.description]: 'I am Marian',
+    });
+  });
+  it('should trigger onChange when the description is changed', () => {
+    render(
+      <EditFreakForm
+        onChange={ onChange }
+      />,
+    );
+
+    const roleInput = screen.getByTestId('role-input');
+
+    fireEvent.change(roleInput, { target: { value: 'Founder' } });
```

Comment:
I think `fireEvent.change` is not a good way to select the 'Founder' role. This is a select element, a user would click on the Select field and then click on one of the options. The `fireEvent.change` helper skips these steps and directly sets the value of the element to the provided value. Tests should be written from the perspective of the user. Ideally we should simulate two clicks, once for displaying the options, and once for selecting one of those options.

Reply:
And this should be done for all other inputs of this kind. I leave it to you imagination to think which ones should be edited in that way.